### PR TITLE
WT-10376 New connection level tiered shared flag to control tiered storage share

### PR DESCRIPTION
**Pull request by kommiharibabu**

### CODE REVIEW

**Comment by ddanderson** [comment]
@kommiharibabu, before approving, I want to understand the usage.  When `shared` is true, this would enable union tables on any newly created tiered table?  And if `shared` is false, we have tiering as we do today?  Or is there something more?  Thanks!

File: src/tiered/tiered_config.c [hunk 1]
```diff
@@ -124,6 +125,8 @@ __wt_tiered_bucket_config(
     WT_ERR(storage->ss_customize_file_system(
       storage, &session->iface, new->bucket, new->auth_token, buf->data, &new->file_system));
     new->storage_source = storage;
+    if (shared.val)
+        new->tiered_shared = true;
```
Comment (sueloverso) [inline src/tiered/tiered_config.c:129]: Is there any testing for this? Given that I think it was in the wrong location in `api_data.py` I think we need a basic usage python test.

**Comment by kommiharibabu** [comment]
@ddanderson Yes, it is correct as you said. When the `shared` flag is false, the tiering logic works how it is done now.

File: dist/api_data.py [hunk 2]
```diff
@@ -205,6 +205,9 @@ def __ge__(self, other):
         Config('object_target_size', '0', r'''
             this option is no longer supported, retained for backward compatibility''',
             min='0', undoc=True),
+        Config('shared', 'true', r'''
```
[the remark this answers is not in the document]
Comment (kommiharibabu) [reply dist/api_data.py]: Thanks. I removed the table-level config as part of this PR. I need to add this option as part of another PR to control the shared option at the table level. I will check other possible ways to pass this option without adding it to the metadata.

File: src/tiered/tiered_config.c (hunk 1 above)
Comment (kommiharibabu) [reply src/tiered/tiered_config.c:129]: Added a basic infrastructure to test the tiered shared and improve it further with additional PR changes.

**Review by ddanderson — APPROVED** [review]
This looks good to me.  As long as @sueloverso is satisfied with the metadata changes, I'm good with this.

**Review by sueloverso — APPROVED** [review]
LGTM. Thank you for moving it to just wiredtiger_open and not into the metadata.

**Comment by kommiharibabu** [comment]
evergreen merge

**Pull request merged**